lib/browser/http.ts

```diff
@@ -236,6 +238,10 @@ export class HttpClientConnection extends BufferedEventEmitter {
     _on_end(stream: HttpClientStream) {
         this.emit('close');
     }
+
+    close() {
+        this.emit('close');
+    }
```

Comment:
This comment doesn't apply to this spot, I'm juts digging into this impl for the first time. I see that it's hooked up so we "end" the connection after it handles 1 response. In the C impl, this would be like an HTTP/1.0 response coming back, or a "connection: close" header happening. In the C impl, we take care that the connection.is_open() returns false during the stream's "end" callback, so that if we hand it back to a connection pool from that callback, the pool knows that the connection is not to be reused.

anyway, we might want to also be sure that `this.axios = null` before invoking the stream's "end" event.

This is a lot of design overhaul feedback for an "improve tests" PR, so we might want to address some of the design differences in a future PR

Reply:
Looks like axios has the ability to "cancel" requests, we might want to use that to cancel requests when the user calls connection.close(), and hold off with the connection's close/end event until those streams finish wrapping up